### PR TITLE
build(.circleci): correct tags filter

### PR DESCRIPTION
**Pull request by leogr**
Signed-off-by: Leonardo Grasso <me@leonardograsso.com>
Co-authored-by: Thomas Labarussias <issif+github@gadz.org>
Signed-off-by: Leonardo Grasso <me@leonardograsso.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

> /area config

> /area outputs

> /area tests

> /area helm

**What this PR does / why we need it**:

According to https://circleci.com/docs/2.0/configuration-reference/

> CircleCI does not run workflows for tags unless you explicitly specify tag filters. Additionally, if a job requires any other jobs (directly or indirectly), you must specify tag filters for those jobs.

This PR just adds explicit filters for release tags.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by Issif — DISMISSED** [review]

**Comment by leogr** [comment]
Just rebased. 
It needs a new approval. cc @Issif 
/hold cancel

**Review by Issif — APPROVED** [review]

**Pull request merged**
